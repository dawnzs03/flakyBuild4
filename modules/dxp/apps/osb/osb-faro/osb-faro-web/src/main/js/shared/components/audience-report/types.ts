export type DataPoint = {
	color: string;
	count: number;
	label: string;
};

export type Dataset = {
	data: DataPoint[];
	empty: Empty;
	total: number;
};

export type Empty = {
	message: string;
	show: boolean;
};

export type TData = {
	audienceReport: {
		anonymousUsersCount: number;
		knownUsersCount: number;
		nonsegmentedKnownUsersCount: number;
		segmentedAnonymousUsersCount: number;
		segmentedKnownUsersCount: number;
	};
	segment: {
		metrics: {
			value: string;
			valueKey: string;
		}[];
		total: number;
	};
};

export enum Name {
	Page = 'page',
	Blog = 'blog',
	Form = 'form',
	Journal = 'journal',
	Document = 'document'
}

export interface IAudienceReportBaseCardProps {
	knownIndividualsTitle: string;
	segmentsTitle?: string;
	uniqueVisitorsTitle?: string;
	query: {
		metricName: string;
		name: Name;
	};
}
