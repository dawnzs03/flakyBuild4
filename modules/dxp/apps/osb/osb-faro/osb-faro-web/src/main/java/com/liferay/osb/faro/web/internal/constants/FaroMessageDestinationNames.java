/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

/**
 * @author Marcos Martins
 */
public class FaroMessageDestinationNames {

	public static final String FARO_CHECK_FARO_PROJECTS_MESSAGE_PROCESSOR =
		"liferay/faro_check_faro_projects_message_processor";

	public static final String FARO_CLEAN_DL_FILE_ENTRY_MESSAGE_PROCESSOR =
		"liferay/faro_clean_dl_file_entry_message_processor";

	public static final String FARO_DAILY_EMAIL_REPORT_MESSAGE_PROCESSOR =
		"liferay/faro_daily_email_report_message_processor";

	public static final String FARO_MONTHLY_EMAIL_REPORT_MESSAGE_PROCESSOR =
		"liferay/faro_monthly_email_report_message_processor";

	public static final String FARO_WEEKLY_EMAIL_REPORT_MESSAGE_PROCESSOR =
		"liferay/faro_weekly_email_report_message_processor";

}