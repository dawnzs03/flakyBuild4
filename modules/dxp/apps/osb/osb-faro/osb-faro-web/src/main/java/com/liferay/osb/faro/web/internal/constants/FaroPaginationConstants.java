/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

import com.liferay.portal.kernel.util.HashMapBuilder;
import com.liferay.portal.util.PropsValues;

import java.util.Map;

/**
 * @author Shinn Lok
 */
public class FaroPaginationConstants {

	public static final int CUR_DEFAULT = 1;

	public static final String ORDER_BY_TYPE_ASC = "asc";

	public static final String ORDER_BY_TYPE_DEFAULT = ORDER_BY_TYPE_ASC;

	public static final String ORDER_BY_TYPE_DESC = "desc";

	public static Map<String, Object> getConstants() {
		return _constants;
	}

	private static final Map<String, Object> _constants =
		HashMapBuilder.<String, Object>put(
			"cur", CUR_DEFAULT
		).put(
			"delta", PropsValues.SEARCH_CONTAINER_PAGE_DEFAULT_DELTA
		).put(
			"deltaValues", PropsValues.SEARCH_CONTAINER_PAGE_DELTA_VALUES
		).put(
			"orderAscending", ORDER_BY_TYPE_ASC
		).put(
			"orderDefault", ORDER_BY_TYPE_DEFAULT
		).put(
			"orderDescending", ORDER_BY_TYPE_DESC
		).build();

}