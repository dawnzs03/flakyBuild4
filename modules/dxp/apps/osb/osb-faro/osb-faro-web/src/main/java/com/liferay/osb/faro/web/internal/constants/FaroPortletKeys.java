/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

/**
 * @author Shinn Lok
 */
public class FaroPortletKeys {

	public static final String CEREBRO = "cerebro_portlet";

	public static final String FARO = "faro_portlet";

}