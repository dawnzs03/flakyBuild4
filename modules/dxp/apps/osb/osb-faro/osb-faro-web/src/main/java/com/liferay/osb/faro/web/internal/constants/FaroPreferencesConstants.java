/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

import com.liferay.portal.kernel.util.HashMapBuilder;

import java.util.Map;

/**
 * @author Matthew Kong
 */
public class FaroPreferencesConstants {

	public static final String SCOPE_GROUP = "group";

	public static final String SCOPE_USER = "user";

	public static Map<String, Object> getScopes() {
		return _scopes;
	}

	private static final Map<String, Object> _scopes =
		HashMapBuilder.<String, Object>put(
			SCOPE_GROUP, SCOPE_GROUP
		).put(
			SCOPE_USER, SCOPE_USER
		).build();

}