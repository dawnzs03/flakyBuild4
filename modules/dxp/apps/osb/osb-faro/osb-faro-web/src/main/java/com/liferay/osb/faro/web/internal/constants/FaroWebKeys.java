/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

/**
 * @author Matthew Kong
 */
public class FaroWebKeys {

	public static final String CEREBRO_ASSETS_URL = "CEREBRO_ASSETS_URL";

	public static final String CEREBRO_TOUCHPOINTS_URL =
		"CEREBRO_TOUCHPOINTS_URL";

}