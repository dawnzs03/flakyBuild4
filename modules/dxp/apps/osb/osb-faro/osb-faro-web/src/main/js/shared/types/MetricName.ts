export enum MetricName {
	Abandonments = 'abandonmentsMetric',
	AnonymousVisitors = 'anonymousVisitorsMetric',
	AvgTimeOnPage = 'avgTimeOnPageMetric',
	BounceRate = 'bounceRateMetric',
	Comments = 'commentsMetric',
	CompletionTime = 'completionTimeMetric',
	Downloads = 'downloadsMetric',
	Entrances = 'entrancesMetric',
	ExitRate = 'exitRateMetric',
	KnownVisitors = 'knownVisitorsMetric',
	Previews = 'previewsMetric',
	Ratings = 'ratingsMetric',
	ReadingTime = 'readingTimeMetric',
	SessionsDuration = 'sessionDurationMetric',
	SessionsPerVisitor = 'sessionsPerVisitorMetric',
	Submissions = 'submissionsMetric',
	Views = 'viewsMetric',
	Visitors = 'visitorsMetric'
}
