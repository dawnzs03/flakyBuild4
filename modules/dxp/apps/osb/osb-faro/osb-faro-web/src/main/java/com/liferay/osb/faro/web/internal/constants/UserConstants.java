/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

import com.liferay.osb.faro.constants.FaroUserConstants;
import com.liferay.portal.kernel.model.RoleConstants;
import com.liferay.portal.kernel.util.HashMapBuilder;

import java.util.Map;

/**
 * @author Matthew Kong
 */
public class UserConstants {

	public static Map<String, String> getRoleNames() {
		return _roleNames;
	}

	public static Map<String, Integer> getStatuses() {
		return _statuses;
	}

	private static final Map<String, String> _roleNames = HashMapBuilder.put(
		"administrator", RoleConstants.SITE_ADMINISTRATOR
	).put(
		"member", RoleConstants.SITE_MEMBER
	).put(
		"owner", RoleConstants.SITE_OWNER
	).build();
	private static final Map<String, Integer> _statuses = HashMapBuilder.put(
		"approved", FaroUserConstants.STATUS_APPROVED
	).put(
		"pending", FaroUserConstants.STATUS_PENDING
	).build();

}