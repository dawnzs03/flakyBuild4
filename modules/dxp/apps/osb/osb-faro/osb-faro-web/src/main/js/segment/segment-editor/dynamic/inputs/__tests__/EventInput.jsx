import * as data from 'test/data';
import EventInput from '../EventInput';
import React from 'react';
import {AttributeTypes} from 'event-analysis/utils/types';
import {createNewGroup} from '../../utils/utils';
import {CustomValue, Property} from 'shared/util/records';
import {fireEvent, render} from '@testing-library/react';
import {fromJS} from 'immutable';
import {MemoryRouter, Route} from 'react-router-dom';
import {MockedProvider} from '@apollo/react-testing';
import {mockEventAttributeDefinitionsReq} from 'test/graphql-data';
import {range} from 'lodash';
import {RelationalOperators} from '../../utils/constants';
import {Routes} from 'shared/util/router';
import {waitForLoading} from 'test/helpers';

jest.unmock('react-dom');

describe('EventInput', () => {
	it('should render', async () => {
		const {container, getAllByRole, getAllByText, getByText} = render(
			<MockedProvider
				mocks={[
					mockEventAttributeDefinitionsReq(
						range(10).map(i =>
							data.mockEventAttributeDefinition(i, {
								__typename: 'EventAttributeDefinition'
							})
						),
						{
							eventDefinitionId: '3',
							page: 0,
							size: 25,
							sort: {
								column: 'name',
								type: 'ASC'
							},
							type: AttributeTypes.Global
						}
					)
				]}
			>
				<MemoryRouter
					initialEntries={[
						'/workspace/23/123123/contacts/segments/create?type=DYNAMIC'
					]}
				>
					<Route path={Routes.CONTACTS_SEGMENT_CREATE}>
						<EventInput
							displayValue='Asset Clicked'
							onChange={jest.fn()}
							operatorRenderer={() => <div>{'test'}</div>}
							property={Property({
								entityName: 'Event',
								id: '3',
								label: 'assetDepthReached',
								name: '3',
								propertyKey: 'event',
								type: 'event'
							})}
							touched={{attribute: true, attributeValue: true}}
							valid={{attribute: true, attributeValue: true}}
							value={CustomValue(
								fromJS({
									criterionGroup: createNewGroup([
										{
											operatorName: 'eq',
											propertyName: 'eventDefinitionId',
											value: '1'
										},
										{
											operatorName: 'contains',
											propertyName: 'attribute/2',
											value: ''
										},
										{
											operatorName: 'gt',
											propertyName: 'day',
											value: 'last24Hours'
										}
									]),
									operator: RelationalOperators.GT,
									value: 1
								})
							)}
						/>
					</Route>
				</MemoryRouter>
			</MockedProvider>
		);

		await waitForLoading(container);

		jest.runAllTimers();

		fireEvent.click(getAllByRole('combobox')[0]);
		fireEvent.click(getByText('since'));
		fireEvent.click(getByText('Last 24 hours'));
		fireEvent.click(getByText('displayName-2'));

		expect(getByText('at least')).toBeTruthy();
		expect(getByText('at most')).toBeTruthy();

		expect(getAllByText('since')[1]).toBeTruthy();
		expect(getByText('after')).toBeTruthy();
		expect(getByText('before')).toBeTruthy();
		expect(getByText('between')).toBeTruthy();
		expect(getByText('ever')).toBeTruthy();
		expect(getByText('on')).toBeTruthy();

		expect(getAllByText('Last 24 hours')[1]).toBeTruthy();
		expect(getByText('Yesterday')).toBeTruthy();
		expect(getByText('Last 7 days')).toBeTruthy();
		expect(getByText('Last 28 days')).toBeTruthy();
		expect(getByText('Last 30 days')).toBeTruthy();
		expect(getByText('Last 90 days')).toBeTruthy();

		expect(getByText('displayName-0')).toBeTruthy();
		expect(getByText('displayName-1')).toBeTruthy();
		expect(getAllByText('displayName-2')[1]).toBeTruthy();
		expect(getByText('displayName-3')).toBeTruthy();
		expect(getByText('displayName-4')).toBeTruthy();
		expect(getByText('displayName-5')).toBeTruthy();
		expect(getByText('displayName-6')).toBeTruthy();
		expect(getByText('displayName-7')).toBeTruthy();
		expect(getByText('displayName-8')).toBeTruthy();
		expect(getByText('displayName-9')).toBeTruthy();

		expect(container).toMatchSnapshot();
	});
});
