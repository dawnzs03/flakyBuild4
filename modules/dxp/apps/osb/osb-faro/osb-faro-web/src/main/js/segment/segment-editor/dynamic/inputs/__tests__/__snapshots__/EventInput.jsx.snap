// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventInput should render 1`] = `
<div>
  <div
    class="criteria-statement"
  >
    <div
      aria-hidden="true"
      class="form-group form-group-autofit"
      data-aria-hidden="true"
    >
      <div
        aria-hidden="true"
        data-aria-hidden="true"
      >
        test
      </div>
      <div
        aria-hidden="true"
        class="form-group-item entity-name form-group-item-label form-group-item-shrink"
        data-aria-hidden="true"
      >
        performed
      </div>
      <div
        aria-hidden="true"
        class="form-group-item display-value form-group-item-label form-group-item-shrink"
        data-aria-hidden="true"
      >
        <b>
          Asset Clicked
        </b>
      </div>
      <div
        aria-hidden="true"
        class="form-group-item form-group-item-shrink"
        data-aria-hidden="true"
      >
        <button
          aria-activedescendant="gt"
          aria-expanded="true"
          aria-haspopup="listbox"
          aria-owns="clay-id-2"
          class="form-control form-control-select form-control-select-secondary operator-input show"
          role="combobox"
          style=""
          tabindex="0"
          type="button"
        />
      </div>
      <div
        aria-hidden="true"
        class="form-group-item form-group-item-shrink"
        data-aria-hidden="true"
      >
        <input
          class="input-root form-control number-input"
          data-testid="occurence-count-input"
          min="0"
          type="number"
          value="1"
        />
      </div>
      <div
        aria-hidden="true"
        class="form-group-item unit form-group-item-label form-group-item-shrink"
        data-aria-hidden="true"
      >
        time(s)
      </div>
      <div
        aria-hidden="true"
        class="form-group-item form-group-item-shrink"
        data-aria-hidden="true"
      >
        <button
          aria-activedescendant="since"
          aria-expanded="true"
          aria-haspopup="listbox"
          aria-owns="clay-id-4"
          class="form-control form-control-select form-control-select-secondary conjunction-input show"
          data-testid="conjunction-input"
          role="combobox"
          style=""
          tabindex="0"
          type="button"
        >
          since
        </button>
      </div>
      <div
        aria-hidden="true"
        class="form-group-item form-group-item-shrink"
        data-aria-hidden="true"
      >
        <button
          aria-activedescendant="last24Hours"
          aria-expanded="true"
          aria-haspopup="listbox"
          aria-owns="clay-id-6"
          class="form-control form-control-select form-control-select-secondary operator-input show"
          data-testid="clay-select"
          role="combobox"
          style=""
          tabindex="0"
          type="button"
        >
          Last 24 hours
        </button>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="form-group form-group-autofit"
      data-aria-hidden="true"
    >
      <div
        aria-hidden="true"
        class="form-group-item conjunction form-group-item-label form-group-item-shrink"
        data-aria-hidden="true"
      >
        where
      </div>
      <div
        class="form-group-item form-group-item-shrink"
      >
        <button
          aria-activedescendant="2"
          aria-expanded="true"
          aria-haspopup="listbox"
          aria-owns="clay-id-8"
          class="form-control form-control-select form-control-select-secondary attribute-input show"
          role="combobox"
          style=""
          tabindex="0"
          type="button"
        >
          displayName-2
        </button>
      </div>
      <div
        aria-hidden="true"
        class="form-group-item form-group-item-shrink"
        data-aria-hidden="true"
      >
        <button
          aria-activedescendant=""
          aria-expanded="false"
          aria-haspopup="listbox"
          class="form-control form-control-select form-control-select-secondary operator-input"
          role="combobox"
          tabindex="0"
          type="button"
        />
      </div>
      <div
        aria-hidden="true"
        class="form-group-item form-group-item-shrink"
        data-aria-hidden="true"
      >
        <input
          class="input-root form-control"
          data-testid="attribute-value-string-input"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
